lib/sycamore/sycamore/transforms/groupby_count.py

```diff
@@ -0,0 +1,107 @@
+from typing import Any, Callable, Optional
+from ray.data import Dataset
+from sycamore.data.document import Document
+from sycamore.plan_nodes import Node, Transform
+
+
+class GroupByCount(Transform):
+    """
+    Count aggregation that allows you to aggregate by document field(s).
+    """
+
+    def __init__(self, child: Node, field: str, unique_field: Optional[str] = None):
+        super().__init__(child)
+        self._field = field
+        self._unique_field = unique_field
+
+    def execute(self, **kwargs) -> "Dataset":
+        # creates dataset
+        ds = self.child().execute(**kwargs)
+
+        # adds a "key" column containing desired field
+        map_fn = self.make_map_fn_count(self._field, self._unique_field)
+        ds = ds.map(map_fn)
+        ds = ds.filter(lambda row: self.filterOutNone(row))
+        # lazy grouping + count aggregation
+
+        if self._unique_field is not None:
+            ds = ds.groupby(["key", "unique"]).count()
+
+        ds = ds.groupby("key").count()
```

Comment:
did we not want to pass kwargs here?

Fwiw we might get rid of the intermediate_datasink but as of now we have it

Reply:
don't think so, since it's a count on grouped data (not a docset) — gives this error `TypeError: GroupedData.count() got an unexpected keyword argument 'write_intermediate_data'`